detectors/aws/ecs/ecs.go

```diff
@@ -101,20 +101,28 @@ func (detector *resourceDetector) Detect(ctx context.Context) (*resource.Resourc
 		if err != nil {
 			return empty, err
 		}
-		attributes = append(
-			attributes,
-			semconv.AWSECSContainerARN(containerMetadata.ContainerARN),
-		)
 
 		taskMetadata, err := ecsmetadata.GetTaskV4(ctx, &http.Client{})
 		if err != nil {
 			return empty, err
 		}
 
-		clusterArn := taskMetadata.Cluster
-		if !strings.HasPrefix(clusterArn, "arn:") {
-			baseArn := containerMetadata.ContainerARN[:strings.LastIndex(containerMetadata.ContainerARN, ":")]
-			clusterArn = fmt.Sprintf("%s:cluster/%s", baseArn, clusterArn)
+		baseArn := detector.getBaseArn(
+			taskMetadata.TaskARN,
+			containerMetadata.ContainerARN,
+			taskMetadata.Cluster,
+		)
+
+		if len(baseArn) > 0 {
```

Comment:
We check string emptiness using `len` here: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/detectors/aws/ecs/ecs.go#L81. Since there is no difference in speed between `len(str) > 0` and `str != ""`, I believe it's better to maintain the code style. WDYT?

Reply:
> I believe it's better to maintain the code style

I agree and `str != ""` is used in all (for sure most) other places (e.g. `gcp` detector). [This line](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/aead73790726a6f6c8cc5facca0c7e43753399aa/detectors/aws/ecs/ecs.go#L81) should be changed as well.